### PR TITLE
Handle tails in Inductor CPU loop splitting

### LINKED ISSUE #195582: [inductor] Silent wrong and non-reproducible results: BatchNorm2d followed by a shrinking F.pad
## 🐛 Describe the bug

A shrinking (negative) `F.pad` placed after a `BatchNorm2d` makes the compiled model return values that are both wrong and **different on every run**, while eager is stable and correct. Magnitudes land around `1e29`–`1e34` against an eager output of about `0.6`, which looks like uninitialized memory being read.

No error, no warning.

## Reproduction

```python
import torch
import torch.nn as nn
import torch.nn.functional as F

class M(nn.Module):
    def __init__(self):
        super().__init__()
        self.conv = nn.Conv2d(1, 50, (2, 2))
        self.bn = nn.BatchNorm2d(50)
        self.fc = nn.Linear(50, 10)

    def forward(self, x):
        x = self.bn(self.conv(x))
        x = x.view(x.shape[0], -1)           # (3, 450)
        x = F.pad(x, (0, 50 - x.shape[-1]))  # pad of -400 -> (3, 50)
        return self.fc(x)

def run(compiled):
    torch.manual_seed(420)
    m = M().eval()
    x = torch.randn(3, 1, 4, 4)
    with torch.no_grad():
        g = torch.compile(m, backend="inductor") if compiled else m
        return g(x).clone()

e1, e2 = run(False), run(False)
print("eager  reproducible:", torch.equal(e1, e2), "| max|out| =", round(e1.abs().max().item(), 4))

outs = []
for _ in range(3):
    torch._dynamo.reset()
    outs.append(run(True))
print("compiled reproducible:", torch.equal(outs[0], outs[1]) and torch.equal(outs[1], outs[2]))
for i, o in enumerate(outs):
    print(f"  compiled run {i}: max|out| = {o.abs().max().item():.6g}")
```

```
--- torch 2.14.0+cpu ---
eager  reproducible: True | max|out| = 0.6061
compiled reproducible: False
  compiled run 0: max|out| = 0.638165
  compiled run 1: max|out| = 1.30965e+34
  compiled run 2: max|out| = 4.51336e+33

--- torch 2.11.0 ---
eager  reproducible: True | max|out| = 0.6061
compiled reproducible: False
  compiled run 0: max|out| = 9.69187e+33
  compiled run 1: max|out| = 5.817e+29
  compiled run 2: max|out| = 5.817e+29
```

Each `run(True)` uses a fresh `torch._dynamo.reset()`; eager is bit-identical across runs in both builds.

## What is and is not needed

Removing either the `BatchNorm2d` or the negative pad makes the compiled output correct and reproducible again:

| model | compiled reproducible | compiled output |
|---|---|---|
| conv → **bn** → view → **negative pad** → fc | **no** | 1e29 … 1e34 |
| conv → bn → relu → view → negative pad → fc | **no** | 1e30 … 1e33 |
| conv → view → negative pad → fc (no bn) | yes | matches eager |
| conv → bn → relu → view → fc (no negative pad) | yes | matches eager |
| view → negative pad → fc (no conv/bn) | yes | matches eager |
| negative pad alone | yes | matches eager |

`relu` is irrelevant, and it happens with the `BatchNorm2d` in either `train()` or `eval()` mode. Both the `BatchNorm2d` and the shrinking pad have to be present.

It is also not gated by the padding configs: the same non-reproducible garbage appears with `TORCHINDUCTOR_INPLACE_PADDING=1`, with `=0`, and with `TORCHINDUCTOR_SHAPE_PADDING` unset.

## Versions

- PyTorch `2.11.0`
- PyTorch `2.14.0+cpu`

CPU, Linux x86_64, Python 3.11.

## Additional context

Checked against nearby padding reports; I do not believe this is a duplicate:

- **#194558** (`in-place padding optimization mishandles negative (shrinking) F.pad … ValueRangeError`) is also about a shrinking `F.pad`, but it fails loudly with a `ValueRangeError` and is gated by `TORCHINDUCTOR_INPLACE_PADDING`. This one is silent, is not affected by that config, and additionally is non-deterministic.
- **#189895** (`constant_pad_nd fused into a split reduction drops the pad's boundary select`, closed) is a **positive** pad on CUDA whose wrong result is deterministic; here the pad is negative, the device is CPU, and the output changes every run.
- **#178680** (`nondeterministic / incorrect results on model with add_, clamp_, squeeze`) is a different set of operators and does not involve padding.

Found by differential testing over StarCoder-generated programs; the reduction above comes from one of them.

cc @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

### PR DESCRIPTION
**Pull request by HusamettinYilmazz**
Fixes #195582

The CPU loop-split optimization replaced an extent N with (N // factor) * factor, silently dropping N % factor iterations. This left output elements unwritten when the loop extent was not divisible by the split factor.

Keep the division-free split for complete blocks and generate an offset unsplit loop for the remainder.

Test Plan:
```bash
lintrunner -a
python test/inductor/test_cpu_repro.py CPUReproTests.test_negative_pad_loop_split_tail -v

```

Passed.

cc @voznesenskym  @penguinwu  @EikanWang  @jgong5  @Guobing-Chen  @XiaobingSuper  @zhuhaozhe  @blzheng  @wenzhe-nrv  @jiayisunx  @ipiszy  @kadeng  @muchulee8  @amjames  @chauhang  @aakhundov  @coconutruben  @jataylo

### CODE REVIEW

**Comment by HusamettinYilmazz** [comment]
Hi @anijain2305  

would you have a chance to take a look at this PR and approve the workflows?
